src/transformers/modeling_longformer.py

```diff
@@ -113,137 +114,167 @@ def __init__(self, config, layer_id):
             attention_window > 0
         ), f"`attention_window` for layer {self.layer_id} has to be positive. Given {attention_window}"
 
-        self.one_sided_attention_window_size = attention_window // 2
+        self.one_sided_attn_window_size = attention_window // 2
 
     @staticmethod
-    def _skew(x, direction):
-        """Convert diagonals into columns (or columns into diagonals depending on `direction`"""
-        x_padded = F.pad(x, direction)  # padding value is not important because it will be overwritten
-        x_padded = x_padded.view(*x_padded.size()[:-2], x_padded.size(-1), x_padded.size(-2))
-        return x_padded
+    def pad_and_transpose_last_two_dims(hidden_states_padded, padding):
+        """Convert diagonals into columns or columns into diagonals depending on `padding`"""
+        hidden_states_padded = F.pad(
+            hidden_states_padded, padding
+        )  # padding value is not important because it will be overwritten
+        hidden_states_padded = hidden_states_padded.view(
```

Comment:
same as `transpose(-1, -2)` That might be clearer.

Reply:
transpose does not work here